### PR TITLE
Exception handlers

### PR DESCRIPTION
**Pull request by abelgalef**
- Added Global Exception Handlers for all the controllers and should be able to map every type of exception (does not map errors). 
- Also added spring dev-tools for hot reload.

### CODE REVIEW

File: src/main/java/miu/edu/resort/controller/GlobalControllerExceptionHandler.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+    // TODO: Review the project and add diffrent error handlers made by others.
+
+    // ALL RUNTIME ERRORS ARE MAPPED TO INTERNAL SERVER ERROR
+    @ExceptionHandler(RuntimeException.class)
+    public ResponseEntity<Object> handleInternalErrors(RuntimeException e) {
+        return new ResponseEntity<>(getErrorMsg(e), HttpStatus.INTERNAL_SERVER_ERROR);
+    }
+
+    // HANDLE 404 ERRORS
```
Comment (karamalqinneh) [inline src/main/java/miu/edu/resort/controller/GlobalControllerExceptionHandler.java:23]: Spring does this out of the box I dont think we need to override it

Comment (abelgalef) [reply src/main/java/miu/edu/resort/controller/GlobalControllerExceptionHandler.java:17]: Yeah, good call, I got rid of the trace in the app.yml too. But we still gotta handle the exceptions when we validate the entities. Lets keep it for now and update it down the line when everyone is finished.

Comment (karamalqinneh) [reply src/main/java/miu/edu/resort/controller/GlobalControllerExceptionHandler.java:17]: Bro you removed the 500 we need that i was talking about the 404 :)

Comment (abelgalef) [reply src/main/java/miu/edu/resort/controller/GlobalControllerExceptionHandler.java:17]: lol... I'll just fix it after my next PR

**Review by karamalqinneh — APPROVED** [review]

**Pull request merged**